src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double INCREASE_CAPACITY = 1.5;
+    private static final int ARRAYS_ZERO_BORDER = 0;
+    private static final int INCREASE_INDEX_FOR_COPY = 1;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (elementData.length == size) {
+            grow();
```

Comment:
we could move this check `if (elementData.length == size)` into `grow` method(this will reduce number if duplicated code). If we do this, then it might be worth to rename `grow` method to `ensureCapacity` or smth like this